api/v1beta2/types.go

```diff
@@ -142,6 +142,21 @@ var (
 	ResourceTypeResourceGroup = ResourceType("resourceGroup")
 )
 
+// COSInstanceReference represents an IBM Cloud COS instance.
+type COSInstanceReference struct {
+	// id is the ID of the COS instance.
+	// +required
+	ID *string `json:"id"`
+
+	// bucketID is the IBM Cloud COS bucket ID.
+	// +optional
+	BucketID *string `json:"bucketID,omitempty"`
```

Comment:
I need to swap back to BucketName, I don't think the IBM COS SDK provides additional details I want to use this.
Regardless of whether I expect CAPI to create COS instances and buckets.